core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -186,6 +186,7 @@ class PartitionTest extends AbstractPartitionTest {
     val leaderEpoch = 10
     val logStartOffset = 0L
     val partition = setupPartitionWithMocks(leaderEpoch = leaderEpoch, isLeader = true)
+    addBrokerEpochToMockMetadataCache(metadataCache, List(remoteReplicaId))
```

Comment:
Hmm, why are these changes needed?

Reply:
Same as above. We use `mock(classOf[MetadataCache])` in `AbstractPartitionTest`.

https://github.com/apache/kafka/blob/94a1bfb1281f06263976b1ba8bba8c5ac5d7f2ce/core/src/test/scala/unit/kafka/cluster/AbstractPartitionTest.scala#L64